nix/magic-folder-repo.nix

```diff
@@ -0,0 +1,9 @@
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.fetchFromGitHub {
+    owner = "LeastAuthority";
+    repo = "magic-folder";
+    rev = "e86068ea8fc9cd7ad0c5d98892d992ccf3115e37";
+    sha256 = "0q34q6pgiv3kzr2jw0551122j72n1dj65h0m194cg4dj0xc89bmp";
```

Comment:
Is it possible to make this float? That does make this derivation not reproducible, but given that the branch this is targeted at does as well, the derivation is more likely to work on the tip of the branch that way.

I guess it would be possible to make the top-level derivation take these as parameters, which would allow *making* it into a reproducible derivation, too. (If all the parameters in this block could be overridden, it would even be straightforward to test against a branch without needing to touch any in-tree code).

Reply:
It looks like `builtins.fetchFromGit` doesn't need a hash. (The things from `pkgs` all do, as they create things in `/nix/store` named after the hash; the ones from `builtins` all have the behavior of caching for 1h).

That said, it looks like the Makefile is now pinning the version of magic-folder. Ideally nix could share that pin, but not a blocker.

Once we are on python3, we could probably use something like https://github.com/pypa/setuptools_scm/pull/580 to get git information even from a github tarball. (Well, we could do it earlier, if we write the code by hand).